cluster/cluster.go

```diff
@@ -52,32 +52,35 @@ const (
 	NoneAuthorizationMode      = "none"
 )
 
-func (c *Cluster) DeployClusterPlanes() error {
-	// Deploy Kubernetes Planes
-	err := services.RunEtcdPlane(c.EtcdHosts, c.Services.Etcd)
-	if err != nil {
+func (c *Cluster) DeployControlPlane() error {
+	// Deploy Etcd Plane
+	if err := services.RunEtcdPlane(c.EtcdHosts, c.Services.Etcd); err != nil {
 		return fmt.Errorf("[etcd] Failed to bring up Etcd Plane: %v", err)
 	}
-	err = services.RunControlPlane(c.ControlPlaneHosts,
+	// Deploy Control plane
+	if err := services.RunControlPlane(c.ControlPlaneHosts,
 		c.EtcdHosts,
 		c.Services,
 		c.SystemImages[ServiceSidekickImage],
 		c.Authorization.Mode,
-		c.HealthcheckDialerFactory)
-	if err != nil {
+		c.HealthcheckDialerFactory); err != nil {
 		return fmt.Errorf("[controlPlane] Failed to bring up Control Plane: %v", err)
 	}
-	err = c.ApplyAuthzResources()
-	if err != nil {
+	// Apply RBAC resources after deploying controlplane
```

Comment:
Nit: not just RBAC, authz configuration in general, even if RBAC is disabled, or another authz mode is enabled.